algo/fleet.go

```diff
@@ -0,0 +1,72 @@
+package algo
+
+import (
+	"errors"
+	"math/rand"
+	"time"
+
+	"github.com/bohdanlisovskyi/Golang-252-Telegram-Bot-Game/database/postgres"
+)
+
+//FleetIsAlive - return true if exists at least one ship with Health > 0, false otherwise
+func FleetIsAlive(fleet []postgres.Ship) bool {
+	for index := range fleet {
+		if fleet[index].Health > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+//ComputeLoadCapacity - return summary load capacity of fleet
+func ComputeLoadCapacity(fleet []postgres.Ship) int {
+	summaryCapacity := 0
+	for _, ship := range fleet {
+		if ship.Health > 0 {
+			summaryCapacity += ship.LoadCapacity
+		}
+	}
+	return summaryCapacity
+}
+
+//ComputeHit - return summary hit of fleet
+func ComputeHit(fleet []postgres.Ship) int {
+	rGen := rand.New(rand.NewSource(time.Now().UnixNano()))
+	summaryHit := 0
+	for _, ship := range fleet {
+		if ship.Health == 0 {
+			continue
+		}
+		if rGen.Float64() <= 0.5 {
+			summaryHit += int(float64(ship.HitRate) * (1.0 + rGen.Float64()*randomRange))
+		} else {
+			summaryHit += int(float64(ship.HitRate) * (1.0 - rGen.Float64()*randomRange))
+		}
+	}
+	return summaryHit
+}
+
+//ComputeFlightTime - return time needed to rich destination between planets
+func ComputeFlightTime(fleet []postgres.Ship, startPlanet, endPlanet *postgres.Planet) (time.Duration, *error) {
```

Comment:
Why it returns pointer *error but not an error?

Reply:
You are right. I will fix it